### DISCUSSION TITLE
A question on timestamps

### DISCUSSION THREAD

**Discussion by webwurm** [AntlerVC/firetable#1102] [Support / Q&A] [answered]
The task is simple, but I can't get it to work:

I want to have a derivative field that has the timestamp of the time, another field changed. But sadly I can't get it to work, not even after reading this [Rowy-blogpost](https://www.rowy.io/blog/firestore-timestamp).

I tried it with
```
const currDate = new Date();
return currDate.getTime();
```
but with this I get a number (which, btw, confuses FlutterFlow so much that it doesn't show any documents in the collection if there you defined it as "timestamp").

I tried it only with 
```
const currTimestamp = new Date()
return currTimestamp
```
but also that doesn't work.

The term
`Timestamp.fromDate(new Date())`
I can't use, because I can't import "Timestamp".

Also
`...FieldValue.serverTimestamp()`
doesn't work.

So what is the right way?

Thanks a lot!

**Comment by webwurm** [comment]
I played again with it - I guess firebase.firestore.Timestamp or firebase.firestore.FieldValue.serverTimestamp() would be correct. But how can I use firebase.firestore in Rowy? Am I just thinking to complicated? I can't figure it out...

**Comment by BeeBombshell** [comment, marked as answer]
Hey @webwurm, you're right. This can be accomplished by using `firebase.firestore.Timestamp`. However, you'd have to manually import the Firestore library like this:
```js
import("@google-cloud/firestore")
```

You could create a derivative field with the output field `Date & Time` and add the following script:
```js
const derivative:Derivative = async ({row,ref,db,storage,auth,logging})=>{
  // WRITE YOUR CODE ONLY BELOW THIS LINE. DO NOT WRITE CODE/COMMENTS OUTSIDE THE FUNCTION BODY
  const date = new Date()
  const ms = date.getTime()
  const { Timestamp } = (await (async () => await import("@google-cloud/firestore"))())
  return Timestamp.fromMillis(ms)
  // WRITE YOUR CODE ONLY ABOVE THIS LINE. DO NOT WRITE CODE/COMMENTS OUTSIDE THE FUNCTION BODY
}
```

### Screenshot of the Column Config
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/53828745/216679172-68180379-43cf-4e25-98dd-8e14e895617d.png">

### Output
<img width="292" alt="image" src="https://user-images.githubusercontent.com/53828745/216679349-bff5c537-4d21-4489-902d-de1f1cb6e47f.png">

**Reply by webwurm** [reply]
Thank you very much, that works perfectly! 
And the best: Now it also works in FlutterFlow - which crashes when there is a number in a field that you define in FlutterFlow as "timestamp"...
